### PR TITLE
[Feature] MTP RL with KL Loss

### PR DESCRIPTION
**Pull request by RangiLyu**

### CODE REVIEW

**Comment by RangiLyu** [comment]
@claude review

File: xtuner/v1/loss/mtp_loss.py [hunk 1]
```diff
@@ -99,12 +141,82 @@ def build(self, data: dict, sp_mesh: DeviceMesh | None = None) -> "MTPLossContex
...
+
+        mtp_logprobs = gather_logprobs(logits, shifted_labels)
+
+        # Use loss_weight from build_batches (globally calibrated) if available,
+        # otherwise fall back to local mask normalization.
+        if loss_weight is not None:
+            kl_weight = loss_weight.flatten()
+        else:
+            mask = (shifted_labels != -100).float()
+            kl_weight = (mask / mask.sum().clamp(min=1)).flatten()
```
Comment (hhaAndroid) [inline xtuner/v1/loss/mtp_loss.py]: 在 sp 情况下，mask.sum 可能是 0，会报错

**Review by hhaAndroid — APPROVED** [review]
